riscv-unix.md

```diff
@@ -43,3 +43,58 @@ be on the same ordering channel.
 
 On RV64I-based Unix-class systems the negative virtual addresses are
 reserved for the kernel.
+
+## Misaligned Physical-Memory Access Atomicity
+
+Consider a data memory access of size *w* to physical address *p<sub>0</sub>*,
+where *w* does not evenly divide *p<sub>0</sub>*.  Let *p<sub>1</sub>* denote
+the physical address of the last byte of the access, and let *P* denote the
+address pair *(p<sub>0</sub>, p<sub>1</sub>)*.  There are two cases:
+
+1. *P* lies within a single physical-memory region.  One of the following
+   holds:
+
+   1. Loads and stores to *P* execute atomically with respect to other
+      accesses to *P*.  AMOs to *P* either execute atomically
+      with respect to other accesses to *P* or raise access
+      exceptions.  LRs and SCs to *P* either execute atomically
+      with respect to other accesses to *P* or raise access exceptions.
+   
+   2. Loads and stores to *P* execute without guarantee of atomicity.  AMOs,
+      LRs, and SCs to *P* raise access exceptions.
+
+2. *P* spans two physical-memory regions. AMOs, LRs, and SCs all raise access
+   exceptions.  Additionally, one of the following holds:
+
+   1. Loads and stores to *P* raise access exceptions.
+
+   2. Loads and stores to *P* succeed without guarantee of atomicity.
+
+   3. Loads and stores to *P* proceed partially, then raise access exceptions.
+      No register writebacks occur.
```

Comment:
This is awkward.  What's the reason for it?

Reply:
This happens when emulating misaligned loads & stores in M-mode software, when the access spans two pages.  The load or store proceeds bytewise but encounters a page fault or access exception once it hits the second page.

This means that a store can complete partially then trap.  For loads, it only matters for memory regions that have side effects.  Usually, this will never happen, since such regions should forbid misaligned accesses altogether.